pkg/gatewayserver/gatewayserver.go

```diff
@@ -433,6 +435,26 @@ type connectionEntry struct {
 	*io.Connection
 	upstreamDone chan struct{}
 	tasksDone    *sync.WaitGroup
+	lastMessage  *ttnpb.UplinkMessage
+}
+
+func isDuplicateUplink(this *ttnpb.UplinkMessage, that *ttnpb.UplinkMessage) bool {
+	switch {
+	case this == nil || that == nil:
```

Comment:
Isn't this an implementation error?

Reply:
`this` will be nil for the first uplink of the connection. `that` will be whatever the frontend sends, I think it's best not to panic here for messages if the frontend is buggy. What do we gain if we skip this?